### PR TITLE
Improve benchmarks of NodeUniqueAttributeConstraintQuery

### PR DESCRIPTION
**Pull request by LucasG0**
Tihs PR adds benchmarks of `NodeUniqueAttributeConstraintQuery` with:
- different uniqueness constraints.
- different neo4j runtime.
- with/without neo4j indexes.

It also adds minor changes:
- neo4j testcontainer exposes http port so we can use neo4j browser while testing.
- `query_benchmark/test_node_manager_query.py` recently introduced has been deleted, as related tests are not useful/relevant.
- `query_analyzer` is not a fixture anymore, currently it seems to make more sense to have it created during `InfrahubDatabaseProfiler` initialization.

### CODE REVIEW

File: backend/tests/query_benchmark/conftest.py [hunk 1]
```diff
@@ -1,44 +1,64 @@
...
+                    },
                 ],
                 "relationships": [
-                    {"name": "owner", "peer": "TestPerson", "optional": True, "cardinality": "one"},
+                    {
+                        "name": "owner",
+                        "label": "Commander of Car",
+                        "peer": "TestPerson",
+                        "cardinality": "one",
+                        "direction": "outbound",
```
Comment (dgarros) [inline backend/tests/query_benchmark/conftest.py]: unless there is a specific reason to set the direction, It's usually best to keep the default

File: backend/infrahub/config.py [hunk 2]
```diff
@@ -192,6 +192,7 @@ class DatabaseSettings(BaseSettings):
     password: str = "admin"
     address: str = "localhost"
     port: int = 7687
+    neo4j_http_port: int = 7474
```
Comment (dgarros) [inline backend/infrahub/config.py]: If it's just for testing purpose, its's best to not add it to the config file
in this case I think we're fine having the value hardcoded for the test

File: backend/tests/conftest.py [hunk 3]
```diff
@@ -120,7 +122,8 @@ async def reset_registry(db: InfrahubDatabase) -> None:
 
 
 @pytest.fixture
-async def default_branch(reset_registry, local_storage_dir, empty_database, db: InfrahubDatabase) -> Branch:
```
Comment (dgarros) [inline backend/tests/conftest.py]: I'm surprised by this change, this fixture is used by hundreds of tests, are you sure of the impact of this change ? and more importantly please can you provide more information why this change is required

File: backend/tests/conftest.py [hunk 4]
```diff
@@ -475,8 +481,6 @@ async def car_person_schema_unregistered(db: InfrahubDatabase, node_group_schema
                         "name": "owner",
                         "label": "Commander of Car",
                         "peer": "TestPerson",
-                        "optional": False,
-                        "kind": "Parent",
```
Comment (dgarros) [inline backend/tests/conftest.py]: please can you provide more information about this change ? why is it related to this PR ?

File: backend/tests/helpers/query_benchmark/db_query_profiler.py [hunk 5]
```diff
@@ -168,9 +181,9 @@ def __exit__(
 
 
 class InfrahubDatabaseProfiler(InfrahubDatabase):
-    def __init__(self, query_analyzer: QueryAnalyzer, **kwargs: Any) -> None:
+    def __init__(self, **kwargs: Any) -> None:
```
Comment (dgarros) [inline backend/tests/helpers/query_benchmark/db_query_profiler.py:154]: I think there is some value to have the query_analyzer decoupled from the InfrahubDatabaseProfiler.
In the future we might want to use the same query_analyzer across multiple tests which might result in multiple InfrahubDatabaseProfiler

**Review by dgarros — COMMENTED** [review]
I'm not sure about the current organization of the tests. I don't think we should separate runtimes, constraints and indexes this way. 

The goal is to be able to support easily any test scenarios.

Also I think we shouldn't assume that we'll use the same dataloader and the same schema for all tests for a given query, the dataloader and the schema should be part of the different dimensions of the tests we can adjust.
I'll schedule some time to discuss that

File: backend/tests/conftest.py (hunk 3 above)
Comment (LucasG0) [reply backend/tests/conftest.py]: I actually did not mean to push these changes, I did that for local testing purposes, nice catch thanks

File: backend/tests/conftest.py (hunk 4 above)
Comment (LucasG0) [reply backend/tests/conftest.py]: Same as above - it should not be there

File: backend/infrahub/config.py (hunk 2 above)
Comment (LucasG0) [reply backend/infrahub/config.py]: Right, this value was actually never read

File: backend/tests/helpers/query_benchmark/db_query_profiler.py (hunk 5 above)
Comment (LucasG0) [reply backend/tests/helpers/query_benchmark/db_query_profiler.py:154]: Yes, actually what is confusing with current design is that `QueryAnalyzer` serves to many purposes:
- it contains the information whether `InfrahubDatabaseProfiler` should profile or not
- it stores measurements
- it generates graphes

I think it would be better that two above tasks are performed directly within `InfrahubDatabaseProfiler`. Thus, `InfrahubDatabaseProfiler` is truly responsible of profiling without needing a `QueryAnalyzer`, and `QueryAnalyzer` could become a `GraphGenerator` responsible only for generating graphes and might be shared accross multiple tests.

Comment (LucasG0) [reply backend/tests/helpers/query_benchmark/db_query_profiler.py:154]: As discussed, will be done in another PR.
